sw/device/silicon_creator/lib/drivers/pinmux.c

```diff
@@ -68,31 +75,54 @@ static const pinmux_input_t kInputSwStrap2 = {
  * @param input A peripheral input and MIO pad to link it to.
  */
 static void configure_input(pinmux_input_t input) {
-  abs_mmio_write32(TOP_EARLGREY_PINMUX_AON_BASE_ADDR +
-                       PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET +
+  abs_mmio_write32(kBase + PINMUX_MIO_PERIPH_INSEL_0_REG_OFFSET +
                        input.periph * sizeof(uint32_t),
                    input.pad);
 }
 
+/**
+ * Enables pull-down for the specified pad.
+ *
+ * @param input A MIO pad.
+ */
+static void enable_pull_down(top_earlgrey_pinmux_insel_t pad) {
+  uint32_t reg =
+      bitfield_bit32_write(0, PINMUX_MIO_PAD_ATTR_0_PULL_EN_0_BIT, true);
+  abs_mmio_write32(
+      kBase + PINMUX_MIO_PAD_ATTR_0_REG_OFFSET + pad * sizeof(uint32_t), reg);
+}
+
+/**
+ * Configures the given pin as input and enables the internal pull-down.
+ *
+ * We enable internal pull-downs since we expect strong pull-ups on the strap
+ * pins.
+ *
+ * @param input A peripheral input and MIO pad to link it to.
+ */
+static void configure_strap_pin(pinmux_input_t input) {
+  configure_input(input);
```

Comment:
I just wanted to double check here.. we are not implementing the full detection algorithm, because we really only care about a singular case, is that right?

So we just enable all the pull-downs, and see if the external world overpowers all of it. If it doesn't for any reason, we don't care what specific value it is since we will not boot_strap anyawys, is that right?

Reply:
That's right, we only care about bootstrap in ROM and we expect strong pull-ups on all sw_strap pins.